src/pages/Main/MainPage.jsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import '../../styles/common.scss';
+import './MainPage.scss';
+import Product from './weeklybest/Product';
+
+class MainPageLayout extends React.Component {
+  state = {
+    Products: [],
+  };
+
+  componentDidMount() {
+    fetch('http://localhost:3000/data/weeklyBest.json', {
```

Comment:
public 폴더에 있는 파일에 접근할때는 
```suggestion
    fetch('/data/weeklyBest.json', {
```
로 적으시면 현재 실행되고 있는 서버 url로 자동적으로 인식됩니다!
오히려 3000포트를 명시해두면, 추후 3001포트, 3002포트 등으로 추가적으로 실행이 된다면 통신이 제대로 안 될 여지가 있겠네요!